patches/chrome-common-BUILD.gn.patch

```diff
@@ -18,7 +18,16 @@ index e6496ae6ef9f22325338c91309444e0bd553e9d8..ea6da1dc85460ee88fca594436dd3f7f
      ":available_offline_content_mojom",
      ":buildflags",
      ":channel_info",
-@@ -542,6 +544,7 @@ static_library("non_code_constants") {
+@@ -257,6 +259,8 @@ static_library("common") {
+     "//url",
+   ]
+ 
++  allow_circular_includes_from = [ "//brave/common" ]
```

Comment:
@bridiver There is a circular dependency between when adding   //chrome/common to //brave/common like below,

```
ERROR Dependency cycle:
  //brave/common:common ->
  //chrome/common:common ->
  //brave/common:common
```

Because, brave_content_client.h has been inheriting ChromeContentClient class.

```
#include "chrome/common/chrome_content_client.h"

class BraveContentClient : public ChromeContentClient {
 public:
  BraveContentClient();
  ~BraveContentClient() override;
```

But, I don't find how to solve this issue yet. Instead, we can allow //brave/common includes //chrome/common by using allow_circular_includes_from. I wonder how do you think about using this. WDYT?

Reply:
@bridiver brave_content_client.h/cc files have already been built separately. So, IMHO, one of ways to avoid the circular dependency is to make BraveContentClient inherit content::ContentClient directly. How do you think about it?

Reply:
@bridiver It looks like I misunderstood what's your comment. I updated this PR again through @simonhong's correction.